### DISCUSSION TITLE
Packet stream issue

### DISCUSSION THREAD

**Discussion by nkciam** [engageub/InternetIncome#190] [Q&A]
![image](https://github.com/user-attachments/assets/92403467-ffad-44aa-aad8-32bbfb113862)
Hi, currently I tried packetstream. Docker container can start and running but when I show the logs it show "Error communicating with PacketStream server". Please help me. Thanks!

**Comment by engageub** [comment]
Hello,
Could you please check if your proxy is working fine.
This usually happens when you are using proxy in 127.0.0.1 format or the server does not respond.
It can also be related to DNS timeouts or the packetstream server itself.
I do see these errors once a while in packetstream logs but not continuous as mentioned in the above screenshot.
If its related to DNS, you will have to replace 8.8.8.8 with 1.1.1.1 and 8.8.4.4 with 1.0.0.1 in the `internetIncome.sh` file.

Thank you

**Reply by nkciam** [reply]
I tried with test branch change 8.8.8.8 with 1.1.1.1 and 8.8.4.4 with 1.0.0.1 in the internetIncome.sh file. Seems traffimonetizer and packetshare not work anymore.

Packetshare logs
![image](https://github.com/user-attachments/assets/b47c7c11-cc22-4b77-915e-267c4a98b1da)


Traffimonetizer logs
![image](https://github.com/user-attachments/assets/dd9c30ad-b24a-433d-b8af-6b3293e742fa)

**Comment by engageub** [comment]
Hello,
Is it working with Direct Connection? It looks like you have enabled SOCKS5 DNS in properties.conf and your proxy does not support.

If you have issues with UDP port on your VPS, you can set the following in test branch to use DNS over HTTPS.
You can leave the other properties to default under Network configuration. 

```
USE_PROXIES=true
USE_DNS_OVER_HTTPS=true
USE_TUN2PROXY=true

```


Thank you

**Reply by engageub** [reply]
Please share tun2socks logs also if you have issues. 
Try with direct connection and then use proxies to confirm if your VPS does not have any issues.

Thank you

**Reply by nkciam** [reply]
![image](https://github.com/user-attachments/assets/133c8e0d-d29a-4641-823c-ed12935c71df)
![image](https://github.com/user-attachments/assets/03a56b67-60d5-4cdd-a4b2-f10b1bdafda8)

USE_PROXIES=true
USE_DNS_OVER_HTTPS=true
USE_TUN2PROXY=true

Tried with tun2proxy but no luck logs
![image](https://github.com/user-attachments/assets/187810c3-0011-4eef-a2d7-1e0d8aed6c46)

**Reply by engageub** [reply]
The error indicates that the DNS requests are being blocked through proxy.
Is your proxy provider blocking the DNS requests ?

On your VPS host, can you run the following commands to check if the DNS is not being blocked with direct connection.

```
nslookup google.com 1.1.1.1
nslookup google.com 8.8.8.8
```

Have you tried with direct connection on your VPS? 
Traffmonetizer accepts DataCenter IPs also.
The other apps usually mention in the logs that the IP is non-residential. 

Thank you

**Reply by nkciam** [reply]
Thank you. I tried again with proxy socks5 and all apps working fine. Http type have issue and I don't know why.

**Reply by engageub** [reply]
> Thank you. I tried again with proxy socks5 and all apps working fine. Http type have issue and I don't know why.

Thank you for the update. May be your proxy provider has issues. Nothing much has been changed in the code recently.

**Comment by TradeBW** [comment]
I face high CPU usage on random VPS from PSclient, Like 3-4K load, when I stop some instances  of Psclient with high cpu then load comes down but when I start them then again it spikes to 3-4K 

Do you know any way to manage such incidences? 

ps -eo pid,pcpu,comm --no-headers | awk '$2 > 1 { printf "PID: %s | CPU: %s%% | CMD: %s\n", $1, $2, $3 }'

I used this to find pids with high cpu.

**Comment by engageub** [comment]
Hi,
PacketStream consumes more resources sometimes which may be due to demand. 
This has already been informed to packetstream support and they mentioned that they would forward to their engineers to investigate. 
I reported it for raspberry PI. You may report this based on your PC configuration.
Please send an email to help@packetstream.io with your CPU architecture details. 


Thank you
